### DISCUSSION TITLE
[Help]: Is 'reInit' event doing something in 'Thumbnails' example?

### DISCUSSION THREAD

**Discussion by Enkratia** [davidjerleke/embla-carousel#735] [Help] [answered]
### Summary

What the purpose of this line in 'Thumbnails' example?: (is this event doing something in the code?)

>     emblaMainApi.on('reInit', onSelect)

Line: 39.
Link is below.

### If applicable, which variants of Embla Carousel are relevant to this question?

- [ ] embla-carousel **(Core)**
- [X] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

### CodeSandbox example

https://codesandbox.io/p/sandbox/embla-carousel-thumbs-react-9nhgr4?file=%2Fsrc%2Fjs%2FEmblaCarousel.tsx%3A39%2C1-39%2C40

**Comment by sarussss** [comment, marked as answer]
When the window is resized, Embla Carousel automatically calls the [reInit](https://www.embla-carousel.com/api/events/#reinit) method which will also fire this event.
So I think it might help in some cases when changing the screen size to avoid errors.

**Reply by davidjerleke** [reply]
Thanks @sarussss 👍! Embla uses `ResizeObserver` instead of a window resize listener from version 8 and up, but still your explanation is correct.
